File: ruby/src/Behavioral/State/DvdStateContext.rb
# //DvdStateContext - the Context

require "Behavioral/State/DvdStateName"
require "Behavioral/State/DvdStateNameStars"

class DvdStateContext

	def initialize() 
		@dvdStateName = DvdStateNameStars.new  # start with stars
	end

	def setDvdStateName (dvdStateNameIn)
		@dvdStateName = dvdStateNameIn
	end

	def showName (nameIn)
		return @dvdStateName.showName(self,nameIn)
	end
end

File: java/src/Behavioral/State/DvdStateContext.java
package Behavioral.State;

//// Original copy of this content taken from http://www.fluffycat.com/Java-Design-Patterns/ in 2010
//// Original Author: Larry Truett
//// Privacy Policy at http://www.fluffycat.com/Privacy-Policy/
//An object appears to change its' class when the class it passes calls through to switches itself for a related class.
//DvdStateContext.java - the Context

public class DvdStateContext {  
   private DvdStateName dvdStateName; 
    
   public DvdStateContext() {
       setDvdStateName(new DvdStateNameStars());  
       //start with stars
   }
   
   public void setDvdStateName(DvdStateName dvdStateNameIn) {
       this.dvdStateName = dvdStateNameIn;
   }
   
   public String showName(String nameIn) {
       return this.dvdStateName.showName(this, nameIn);
   }
}
